gix-fs/tests/stack/mod.rs

```diff
@@ -35,97 +35,97 @@ fn p(s: &str) -> &Path {
 #[test]
 #[cfg(windows)]
 fn path_join_handling() {
-    let absolute = p("/absolute");
+    let looks_absolute = p("/absolute");
     assert!(
-        absolute.is_relative(),
-        "on Windows, absolute Linux paths are considered relative (and relative to the current drive)"
+        looks_absolute.is_relative(),
+        "on Windows, 'absolute' Linux paths are relative (and relative to the current drive)"
     );
-    let bs_absolute = p("\\absolute");
+    let bs_looks_absolute = p(r"\absolute");
     assert!(
-        absolute.is_relative(),
+        bs_looks_absolute.is_relative(),
         "on Windows, strange single-backslash paths are relative (and relative to the current drive)"
     );
     assert_eq!(
-        p("relative").join(absolute),
-        absolute,
-        "relative + absolute = absolute - however, they kind of act like they are absolute in conjunction with relative base paths"
+        p("relative").join(looks_absolute),
+        looks_absolute,
+        "relative + unix-absolute = unix-absolute - the relative path without a drive is replaced"
     );
     assert_eq!(
-        p("relative").join(bs_absolute),
-        bs_absolute,
-        "relative + absolute = absolute - backslashes aren't special here, and it just acts like it's absolute"
+        p("relative").join(bs_looks_absolute),
+        bs_looks_absolute,
+        "relative + unix-absolute = unix-absolute - the relative path without a drive is replaced - backslashes aren't special here"
     );
 
     assert_eq!(
         p("c:").join("relative"),
         p("c:relative"),
-        "drive + relative = strange joined result with missing backslash, but it's a valid path that works just like `c:\relative`"
+        "drive + relative = relative to the drive-specific current directory"
```

Comment:
This change relates to https://github.com/Byron/gitoxide/pull/1374#discussion_r1721045107.